### PR TITLE
Convert DOMDocument warnings into exceptions when validating

### PR DESCRIPTION
**Pull request by ziadoz**

### CODE REVIEW

File: src/Xml.php [hunk 1]
```diff
@@ -42,6 +43,10 @@ public static function validate($filePath, $schemaPath)
 
         libxml_use_internal_errors(true);
 
+        set_error_handler(function ($number, $message, $file, $line, $context) {
+            throw new PangaeaException($message, $number);
```
Comment (gabriel403) [inline src/Xml.php]: Would `$message, $number, $number, $file, $line` give more context?

File: tests/ItemsTest.php [hunk 2]
```diff
@@ -94,6 +94,18 @@ public function testInvalidLifecycleStatusException()
...
+    /**
+     * @expectedException               \Pangaea\PangaeaException
+     * @expectedExceptionMessageRegExp  /Entity 'bull' not defined/
+     */
+    public function testProductsInvalidEntitiesException()
+    {
+        $this->item->setTitle('product number 123 &amp;bull;');
+        $this->feed->addItem($this->item);
+
+        $this->assertTrue($this->feed->save(__DIR__ . '/output/items.xml'));
```
Comment (gabriel403) [inline tests/ItemsTest.php]: Why assert anything? You're testing exceptions?

File: src/Xml.php (hunk 1 above)
Comment (ziadoz) [inline src/Xml.php]: I've changed them to match what the PHP manual refers to them as. Hopefully that's a bit clearer.

File: tests/ItemsTest.php (hunk 2 above)
Comment (ziadoz) [inline tests/ItemsTest.php]: Ta. I've removed this.

**Comment by gabriel403** [comment]
:+1:

**Pull request merged**